makelib/golang.mk

```diff
@@ -241,21 +263,36 @@ go.vendor: $(DEP)
 	@$(DEP) ensure || $(FAIL)
 	@$(OK) dep ensure
 
+go.mod.verify:
+	@$(INFO) verify dependencies have expected content
+	@$(GO) mod verify || $(FAIL)
+	@$(OK) go modules dependencies verified
+
+go.mod.update:
+	@$(INFO) update go modules
+	@$(GO) get -u ./... || $(FAIL)
+	@$(OK) update go modules
+
+go.mod.vendor:
+	@$(INFO) go mod vendor
+	@$(GO) mod vendor || $(FAIL)
+	@$(OK) go mod vendor
+
 go.clean:
+	@# make all read-only go modules folder writable, so they can be deleted
+	@if [ -d $(GO_MOD_DIR) ]; then \
+		find $(GO_MOD_DIR) -wholename '*pkg/mod' | \
+			while read folder ; do find $$folder -type d -exec chmod +w {} \; \
+		; done \
+	fi
+
 	@rm -fr $(GO_BIN_DIR) $(GO_TEST_DIR)
 
 go.distclean:
 	@rm -rf $(GO_VENDOR_DIR) $(GO_PKG_DIR)
 
-go.generate: $(GOIMPORTS)
-	@$(INFO) go generate $(PLATFORM)
-	@CGO_ENABLED=0 $(GOHOST) generate $(GO_COMMON_FLAGS) $(GO_PACKAGES) $(GO_INTEGRATION_TEST_PACKAGES) || $(FAIL)
-	@$(OK) go generate $(PLATFORM)
-	@find $(GO_SUBDIRS) $(GO_INTEGRATION_TESTS_SUBDIRS) -type f -name 'zz_generated*' -exec $(GOIMPORTS) -l -w -local $(GO_PROJECT) {} \;
-
```

Comment:
I think there are cases where `go generate` outside of kubebuilder scenarios. does this need to be removed?

Reply:
I didn't see a response to this one and it was resolved.

Reply:
I put back the `go.generate` target as you suggested, and then resolved. Do you have more suggestions?